src/client/utils.js

```diff
@@ -75,13 +77,13 @@ export function clearCookie(name) {
 // Tries the JWT from the given key either from a storage or the cookie
 export function getJWT(tokenKey, cookieKey, storage) {
   return Promise.resolve(storage.getItem(tokenKey)).then(jwt => {
-    const cookieToken = getCookie(cookieKey);
-
-    if (cookieToken) {
-      return cookieToken;
+    let token = jwt || getCookie(cookieKey);
+    if (token) {
+      if (decode(token).exp * 1000 < new Date().getTime()) {
```

Comment:
can we wrap this in a separate `isExpired` function or something? Might be handy to use outside of just the `getJWT` call on the client.